packages/debug/src/browser/debug-session.tsx

```diff
@@ -304,8 +324,17 @@ export class DebugSession implements CompositeTreeElement {
         }
     }
 
+    /**
+     * The `send('initialize')` request could resolve later than `on('initialized')` emits the event.
```

Comment:
This seems odd: doesn't https://microsoft.github.io/debug-adapter-protocol/specification#Events_Initialized state that the `initialize` request has been handled before the `initialized` event is sent? How would the event overtake the reply? Are we handling incorrect debug adapter implementations here?

Reply:
> Are we handling incorrect debug adapter implementations here?

Yes, on the spot. Please reference https://github.com/eclipse-theia/theia/issues/11886 describing the nature of the problem.